src/test/regress/specs/isolation_insert_vs_all.spec

```diff
@@ -0,0 +1,83 @@
+step "s1-distribute-table" { SELECT create_distributed_table('insert_hash', 'id'); }
+step "s1-select-count" { SELECT COUNT(*) FROM insert_hash; }
+step "s1-commit" { COMMIT; }
+
+# session 2
+session "s2"
+step "s2-insert" { INSERT INTO insert_hash VALUES(7, 'k'); }
+step "s2-update" { UPDATE insert_hash SET data = 'l' WHERE id = 4; }
+step "s2-delete" { DELETE FROM insert_hash WHERE id = 4; }
+step "s2-truncate" { TRUNCATE insert_hash; }
+step "s2-drop" { DROP TABLE insert_hash; }
+step "s2-ddl-create-index" { CREATE INDEX insert_hash_index ON insert_hash(id); }
+step "s2-ddl-drop-index" { DROP INDEX insert_hash_index; }
+step "s2-ddl-create-index-concurrently" { CREATE INDEX CONCURRENTLY insert_hash_index ON insert_hash(id); }
+step "s2-ddl-add-column" { ALTER TABLE insert_hash ADD new_column int DEFAULT 0; }
+step "s2-ddl-drop-column" { ALTER TABLE insert_hash DROP new_column; }
+step "s2-ddl-rename-column" { ALTER TABLE insert_hash RENAME data TO new_data; }
+step "s2-table-size" { SELECT citus_table_size('insert_hash'); SELECT citus_relation_size('insert_hash'); SELECT citus_total_relation_size('insert_hash'); }
+step "s2-master-modify-multiple-shards" { SELECT master_modify_multiple_shards('DELETE FROM insert_hash;'); }
+step "s2-create-non-distributed-table" { CREATE TABLE insert_hash(id integer, data text); COPY insert_hash FROM PROGRAM 'echo 0, a\\n1, b\\n2, c\\n3, d\\n4, e' WITH CSV; }
+step "s2-distribute-table" { SELECT create_distributed_table('insert_hash', 'id'); }
+step "s2-select" { SELECT * FROM insert_hash ORDER BY id, data; }
+
+# permutations - INSERT vs INSERT
+permutation "s1-begin" "s1-insert" "s2-insert" "s1-commit" "s1-select-count"
+
+# permutations - INSERT first
+permutation "s1-begin" "s1-insert" "s2-update" "s1-commit" "s1-select-count"
+permutation "s1-begin" "s1-insert" "s2-delete" "s1-commit" "s1-select-count"
+permutation "s1-begin" "s1-insert" "s2-truncate" "s1-commit" "s1-select-count"
+permutation "s1-begin" "s1-insert" "s2-drop" "s1-commit" "s1-select-count"
+permutation "s1-begin" "s1-insert" "s2-ddl-create-index" "s1-commit" "s1-select-count"
+permutation "s1-ddl-create-index" "s1-begin" "s1-insert" "s2-ddl-drop-index" "s1-commit" "s1-select-count"
+permutation "s1-begin" "s1-insert" "s2-ddl-create-index-concurrently" "s1-commit" "s1-select-count"
+permutation "s1-begin" "s1-insert" "s2-ddl-add-column" "s1-commit" "s1-select-count"
+permutation "s1-ddl-add-column" "s1-begin" "s1-insert" "s2-ddl-drop-column" "s1-commit" "s1-select-count"
+permutation "s1-begin" "s1-insert" "s2-table-size" "s1-commit" "s1-select-count"
+permutation "s1-begin" "s1-insert" "s2-master-modify-multiple-shards" "s1-commit" "s1-select-count"
+permutation "s1-drop" "s1-create-non-distributed-table" "s1-begin" "s1-insert" "s2-distribute-table" "s1-commit" "s1-select-count"
```

Comment:
Hmmm, this was not on the test document but I can add it as a separate test in part 2 of this PR.